gcloud-java-pubsub/src/test/java/com/google/cloud/pubsub/TopicTest.java

```diff
@@ -0,0 +1,320 @@
+/*
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.pubsub;
+
+import static org.easymock.EasyMock.createMock;
+import static org.easymock.EasyMock.createStrictMock;
+import static org.easymock.EasyMock.expect;
+import static org.easymock.EasyMock.replay;
+import static org.easymock.EasyMock.verify;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+
+import com.google.cloud.AsyncPage;
+import com.google.cloud.AsyncPageImpl;
+import com.google.cloud.Page;
+import com.google.cloud.PageImpl;
+import com.google.cloud.pubsub.PubSub.ListOption;
+import com.google.common.collect.ImmutableList;
+import com.google.common.util.concurrent.Futures;
+
+import org.junit.After;
+import org.junit.Test;
+
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+
+public class TopicTest {
+
+  private static final String NAME = "topic";
+  private static final TopicInfo TOPIC_INFO = TopicInfo.of(NAME);
+
+  private PubSub serviceMockReturnsOptions = createStrictMock(PubSub.class);
+  private PubSubOptions mockOptions = createMock(PubSubOptions.class);
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Private field 'mockOptions' could be made final; it is only initialized in the declaration or constructor.](https://www.codacy.com/app/mziccard/gcloud-java/file/2613573683/issues/source?bid=3123109&fileBranchId=3311984#l49)